src/eredis_client.erl

```diff
@@ -41,7 +41,7 @@
 -record(state, {
```

Comment:
I guess eredis_client does not support code_change? Because if it does then the change in state would need to be handled I suppose?

Reply:
Good point! I think a code_change that turns an auth_cmd which is iodata into a fun might do...

Reply:
We've added `sentinel` to the state. It breaks code_change too, between versions before and after the addition of sentinel... @e-kmisk Do you need state conversion code for every version change?

Reply:
I've update `code_change/3` so that it succeeds only if the state record has not changed. Otherwise it fails with `{error, not_implemented}` which blocks the hot upgrade rather than crashing later. Better?

Reply:
Seems ok to me.

Reply:
Good, so I consider this ready to merge. Agree?